docs/diagnostics/PX1015.md

```diff
@@ -0,0 +1,45 @@
+# PX1015
+This document describes the PX1015 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                                                                                                      | Type                           | Code Fix    | 
+| ------ | -------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------ | ----------- | 
+| PX1015 | For a BQL statement that contains parameters, the number of arguments of a `Select` method is different from the number of parameters. | Warning (Level 1: Significant) | Unavailable | 
+
+## Diagnostic Description
+For BQL statements that contain `Required` and `Optional` parameters, a `Select` method must have a number of arguments that satisfies both of the following requirements:
+ - The number of arguments is equal to or greater than the number of `Required` parameters in the BQL statement.
+ - The number of arguments is equal to or less than the number of `Required` and `Optional` parameters in the BQL statement.
+
+To fix the issue, you should check whether all necessary arguments are passed to the method.
+
+The diagnostic supports the following method calls:
+
+ - The family of `Select` methods, such as `Select`, `SelectSingleBound`, and `SelectWindowed`, of the standard `PXSelect` classes
+ - The family of `Search` methods of the standard `PXSelect` classes
+ - The `Update` method of the `PXUpdate` classes
+ - The static `Select` method of BQL classes derived from the standard `PXSelect` classes
+
+The diagnostic does not support the `Select` method calls for instances of BQL classes derived from the standard `PXSelect` classes.
```

Comment:
@k-s-popova It's worth noting that the diagnostic doesn't support the following cases:
- dynamicaly constructed BQL via `WhereAnd`, Compose and similar methods
- cases when it's impossible to analyze BQL at compile time

Reply:
@SENya1990, "cases when it's impossible to analyze BQL at compile time" - can you provide an example of such case?

Reply:
@k-s-popova sure:
```
public IEnumerable  MakeSelect(PXSelectBase query, object[] args)
{
   return query.Select(args);
}
```
and 
```
PXSelectBase query = new PXSelect<TDac,
                                                Where<TDac.id, Equal<Required<TDac.id>>>>(graph);

if (someCondition)
{
  query = new PXSelect<TDac,
                                 Where<TDac.id, Equal<Required<TDac.id>>,
                                        And<TDac.type, Equal<Required<TDac.type>>>>>(graph);
}

query.Select(id, type);
```

Reply:
@k-s-popova also the arguments also not always could be analyzed by the diagnostic. Consider the following:
```
object[] args = new[] { id };

if (type != null)
{
   args  = new [] { id, type };
}

var res = PXSelect<TDac,
                 Where<TDac.id, Equal<Required<TDac.id>>,
                       And<TDac.type, Equal<Optional<TDac.type>>>>>
                .Select(graph, currents: null, pars: args);
```